### PR TITLE
docs: add v0.3 AWS guide

### PR DESCRIPTION
**Pull request by andrewrynhard**
This adds documentation for v0.3 AWS users.

### CODE REVIEW

File: docs/website/content/v0.3/en/guides/cloud/aws.md [hunk 1]
```diff
@@ -2,18 +2,225 @@
...
+```bash
+aws s3api create-bucket \
+    --bucket talos-aws-tutorial-${ACCOUNT} \
+    --create-bucket-configuration LocationConstraint=${REGION} \
+    --acl private
+```
+
+##### Create the `vmimport` Role
+
+In order to create an AMI, ensure that the `vmimport` role exists as described in the [official AWS documentation](https://docs.aws.amazon.com/vm-import/latest/userguide/vmie_prereqs.html#vmimport-role).
```
Comment (bradbeam) [inline docs/website/content/v0.3/en/guides/cloud/aws.md]: May want to add a note / sentence saying that this role should be associated with the bucket created above ( `talos-aws-tutorial-${ACCOUNT}` ) just to highlight/reiterate the purpose.

File: docs/website/content/v0.3/en/guides/cloud/aws.md [hunk 2]
```diff
@@ -2,18 +2,225 @@
...
-Once the AMI is created, you can now start an EC2 instance using the AMI ID.
-Provide the proper configuration as the instance's user data.
+Once the `SnapshotTaskDetail.Status` indicates `completed`, we can register the image.
+
+##### Register the Image
+
+```bash
+aws ec2 register-image \
+    --region ${REGION} \
+    --block-device-mappings "DeviceName=/dev/xvda,VirtualName=talos,Ebs={DeleteOnTermination=true,SnapshotId=${SNAPSHOT},VolumeSize=4,VolumeType=gp2}" \
```
Comment (bradbeam) [inline docs/website/content/v0.3/en/guides/cloud/aws.md]: Should we have a variable for `VolumeSize`?

File: docs/website/content/v0.3/en/guides/cloud/aws.md [hunk 3]
```diff
@@ -2,18 +2,225 @@
...
+## Creating a Cluster via the AWS CLI
+
+In this guide we will create an HA Kubernetes cluster with 3 worker nodes.
+We assume an existing VPC.
+
+### Create the Subnet
+
+```bash
+aws ec2 create-subnet \
+    --region ${REGION} \
```
Comment (smira) [inline docs/website/content/v0.3/en/guides/cloud/aws.md]: should we explain what should be the values for all the vars/provide some samples?

Comment (andrewrynhard) [reply docs/website/content/v0.3/en/guides/cloud/aws.md]: Examples would be good 👍.

File: docs/website/content/v0.3/en/guides/cloud/aws.md [hunk 4]
```diff
@@ -2,18 +2,225 @@
...
+##### Create the Image Snapshot
+
+```bash
+curl -LO https://github.com/talos-systems/talos/releases/download/v0.3.0-alpha.2/aws.tar.gz
+tar -xvf aws.tar.gz
+aws s3 cp aws.raw s3://${BUCKET}/talos-aws-tutorial.raw
+aws ec2 import-snapshot \
+    --region ${REGION} \
+    --description "Talos kubernetes tutorial" \
+    --disk-container "Format=raw,UserBucket={S3Bucket=${BUCKET},S3Key=talos-aws-tutorial.raw}"
```
Comment (smira) [inline docs/website/content/v0.3/en/guides/cloud/aws.md]: looks like `$BUCKET` should be  `talos-aws-tutorial-${ACCOUNT}` from the above?

File: docs/website/content/v0.3/en/guides/cloud/aws.md [hunk 5]
```diff
@@ -2,18 +2,225 @@
...
+
+### Configure the Load Balancer
+
+```bash
+aws elbv2 create-target-group \
+    --region ${REGION} \
+    --name talos-aws-tutorial-tg \
+    --protocol TCP \
+    --port 6443 \
+    --vpc-id ${VPC}
```
Comment (bradbeam) [inline docs/website/content/v0.3/en/guides/cloud/aws.md]: do we want to add a target group for `tcp/50000` yet in prep for the upcoming changes?

File: docs/website/content/v0.3/en/guides/cloud/aws.md [hunk 6]
```diff
@@ -2,18 +2,225 @@
...
+    --region ${REGION} \
+    --block-device-mappings "DeviceName=/dev/xvda,VirtualName=talos,Ebs={DeleteOnTermination=true,SnapshotId=${SNAPSHOT},VolumeSize=4,VolumeType=gp2}" \
+    --root-device-name /dev/xvda \
+    --virtualization-type hvm \
+    --architecture x86_64 \
+    --ena-support \
+    --name talos-aws-tutorial-ami
+```
+
+We now have an AMI we can use to create our cluster.
```
Comment (smira) [inline docs/website/content/v0.3/en/guides/cloud/aws.md]: save AMI id in `$AMI` ?

File: docs/website/content/v0.3/en/guides/cloud/aws.md (hunk 2 above)
Comment (andrewrynhard) [reply docs/website/content/v0.3/en/guides/cloud/aws.md]: In reality they only need ~100Mb to fit the Talos install. once a user creates an AMI, the disk will expand to the volume size the create the instance with.

File: docs/website/content/v0.3/en/guides/cloud/aws.md (hunk 5 above)
Comment (andrewrynhard) [reply docs/website/content/v0.3/en/guides/cloud/aws.md]: I don't think so. I thought we decided on client-side load balancing?

File: docs/website/content/v0.3/en/guides/cloud/aws.md (hunk 6 above)
Comment (andrewrynhard) [reply docs/website/content/v0.3/en/guides/cloud/aws.md]: Yeah I'm realizing that I need to explain when to save what.

File: docs/website/content/v0.3/en/guides/cloud/aws.md (hunk 1 above)
Comment (andrewrynhard) [reply docs/website/content/v0.3/en/guides/cloud/aws.md]: Good point.

File: docs/website/content/v0.3/en/guides/cloud/aws.md (hunk 4 above)
Comment (andrewrynhard) [reply docs/website/content/v0.3/en/guides/cloud/aws.md]: Good catch. I think I want to keep it generic so I will stick with `$BUCKET`

**Pull request merged**